tests/test_2fa_duo.py

```diff
@@ -3,264 +3,205 @@
+            self.browser.send_inputs(netid2, self.password)
+            self.browser.click(Locators.submit_button)
+            self.browser.wait_for_tag('p', "Using your 'sptest03' Duo identity.")
+            enter_duo_passcode(self.browser, match_service_provider=self.sp)
+
+    def test_2fa_session_multiple_crn(self, netid10, enter_duo_passcode):
         """
         2FA-8 part b. 2FA session using an acct with multiple CRNs.
         """
         # b. For primary authn, use sptest10 (no 2FA account but with a CRN list of sptest03, sptest07).
-        fresh_browser = Chrome()
         with self.utils.using_test_sp(self.sp):
-            fresh_browser.get(f'{self.sp_url(self.sp)}/shib{self.test_env}mfa')
-            fresh_browser.send_inputs(netid10, self.password)
-            fresh_browser.click(Locators.submit_button)
-            fresh_browser.wait_for_tag('div', 'Select a UW NetID for 2nd factor authentication.')
-            fresh_browser.find_element_by_xpath("//input[@value='sptest07']").click()
-            fresh_browser.snap()
-            fresh_browser.click(Locators.submit_button)
-            fresh_browser.wait_for_tag('p', 'Use your 2FA device.')
-            fresh_browser.wait_for_tag('p', "Using your 'sptest07' Duo identity.")
-
-            fresh_browser.switch_to.frame(fresh_browser.find_element_by_id('duo_iframe'))
-            fresh_browser.execute_script("document.getElementById('passcode').click()")
-            fresh_browser.execute_script("document.getElementsByClassName('passcode-input')[0].value=arguments[0]", self.passcode)
-            fresh_browser.snap()
-            fresh_browser.execute_script("document.getElementById('passcode').click()")
-            fresh_browser.wait_for_tag('h2', f'{self.sp_domain(self.sp)} sign-in success!')
-            fresh_browser.close()
-
-
-def test_remember_me_cookie(utils, sp_url, sp_domain, secrets, netid3, test_env):
+            self.browser.get(self.shib_mfa_url)
+            self.browser.send_inputs(netid10, self.password)
+            self.browser.click(Locators.submit_button)
+            self.browser.wait_for_tag('div', 'Select a UW NetID for 2nd factor authentication.')
+            self.browser.find_element_by_xpath("//input[@value='sptest07']").click()
+            self.browser.snap()
+            self.browser.click(Locators.submit_button)
+            self.browser.wait_for_tag('p', "Using your 'sptest07' Duo identity.")
+            enter_duo_passcode(self.browser, match_service_provider=self.sp)
+
+
+def test_remember_me_cookie(
+        utils, sp_shib_url, sp_url, log_in_netid,
+        sp_domain, secrets, netid3, test_env, fresh_browser, enter_duo_passcode):
     """
     2FA-9 Remember me cookie
     """
-    # chrome_options = webdriver.ChromeOptions()
-    # chrome_options.add_experimental_option("prefs", {"profile.default_content_settings.cookies": 1})
-    # fresh_browser = Chrome(options=chrome_options)
-    fresh_browser = Chrome()
+    fresh_browser
```

Comment:
Does `fresh_browser` have a value?

Reply:
of course it has a value, it's a fixture, but no, this line shouldn't be here.